controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -760,15 +761,18 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
                 .thenCompose(configProperty -> {
                     // 2. post event to start update workflow
                     if (!configProperty.getObject().isUpdating()) {
-                        return eventHelperFuture.thenCompose(eventHelper -> eventHelper.addIndexAndSubmitTask(
-                                new UpdateStreamEvent(scope, stream, requestId),
-                                // 3. update new configuration in the store with updating flag = true
-                                // if attempt to update fails, we bail out with no harm done
-                                () -> streamMetadataStore.startUpdateConfiguration(scope, stream, newConfig,
-                                        context, executor))
-                                // 4. wait for update to complete
-                                .thenCompose(x -> eventHelper.checkDone(() -> isUpdated(scope, stream, newConfig, context))
-                                        .thenApply(y -> UpdateStreamStatus.Status.SUCCESS)));
+                        Set<String> currentTags = new HashSet<>(configProperty.getObject().getStreamConfiguration().getTags());
+                        currentTags.removeAll(newConfig.getTags());
+                        return removeTagIndexEntries(scope, stream, currentTags, context ).thenCompose(v ->
```

Comment:
this should be done as part of the workflow so that we dont fail in between the removal and have only partial removal.

Reply:
fixed it.